keps/sig-node/3008-cri-class-based-resources/README.md

```diff
@@ -0,0 +1,1504 @@
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+We would like to add support for QoS-class resources in Kubernetes.
+QoS-class resources can be thought of as non-accountable resources, each of
+which is presented by a set of classes. Being non-accountable means that
+multiple containers can be assigned to the same class. They are also supposed
+to be opaque to the CRI client in the sense that the container runtime takes
+care of configuration and control of the resources and the classes within.
+
+A prime example of a QoS-class resource is Intel RDT (Resource Director
+Technology). RDT is a technology for controlling the cache lines and memory
+bandwidth available to applications. RDT provides a class-based approach for
+QoS control of these shared resources: all processes in the same hardware class
+share a portion of cache lines and memory bandwidth.
+
+We also believe that the Linux Block IO controller (cgroup) should be handled
+as a QoS-class resource on the level of container orchestration. This enables
+configuring I/O scheduler priority and throttling I/O bandwidth per workload.
+Having the support for QoS-class resources in place, it will provide a
+framework for the future, for instance class-based network or memory type
+prioritization.
+
+We have identified the need for both container-level and pod-level QoS-class
+resources as independent concepts. Intel RDT (above) is per-container by
+design because of the hardware implementation (the control/class hiearchy is
+flat). Also, the current support for blockio is container-level only (it is not
+possible to configure pod sandbox-level cgroup parameters). However, we have
+plans for implementing configuration of sandbox-level blockio parameters. Other
+usage for pod sandbox-level QoS-class resources would be communicating the
+Kubernetes Pod QoS class from kubelet to the container runtime.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+RDT implements a class-based mechanism for controlling the cache and memory
```

Comment:
What's RDT?  Maybe hyperlink to it, or repeat the explanation from earlier.

Reply:
Good point. I added a hyperlink (and for the linux resctrl interface, too)